### PR TITLE
Add documentation for nonroot Docker variant

### LINKED ISSUE #409: Add documentation for nonroot Docker variant
I just noticed the new option of having a non-root Docker container as introduced by #357.

Unfortunately I couldn't find any documentation on it, especially any required manual migration steps as mentioned in https://github.com/almarklein/timetagger/pull/357#issuecomment-1685026399.

I'd be willing to help with writing the documentation but it would probably be more efficient to get a head start by knowing what we're talking about here. Just `chown`-ing the data directory? Any pointers here @danielhass?

### PR DESCRIPTION
**Pull request by kraftner**
Ongoing work to fix #409

### CODE REVIEW

**Review by danielhass — COMMENTED** [review]
@kraftner I think you left out the migration instructions on purpose for now?

I was able to migrate an instance on my side again and it seems that we got all the steps needed as described over in #409.

@almarklein what do you think, shall we maybe move a lot of the details around the Docker-based hosting currently present in this article https://timetagger.app/articles/selfhost2/ over to a "proper" docs page?

File: README.md [hunk 1]
```diff
@@ -61,7 +61,7 @@ If the server runs on your local machine, you can use single-user mode out-of-th
 
 A docker image is provided via the [Github container registry](https://github.com/almarklein/timetagger/pkgs/container/timetagger), so you can use e.g. Docker-compose to easily host your
 own server.
-See the example [docker-compose.yml](https://github.com/almarklein/timetagger/blob/main/deploy/docker-compose.yml).
```
[the remark this answers is not in the document]
Comment (almarklein) [reply README.md:64]: I suppose anyone reading this section will wonder what the difference is between root/nonroot. So perhaps this should eventually just be a link to a docs page on the (different) docker images. Until then it makes sense to add a brief explainer here, I suppose.

**Review by almarklein — COMMENTED** [review]
> what do you think, shall we maybe move a lot of the details around the Docker-based hosting currently present in this article https://timetagger.app/articles/selfhost2/ over to a "proper" docs page?

Yeah, having a page in the docs with details about the docker images and how to use them sounds like a good plan. Then we can there take the space to explain the difference between root and nonroot, migration, etc. Does not have to be in this PR.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (kraftner) [reply README.md:64]: https://github.com/almarklein/timetagger/pull/410/commits/c1c394c163a812db62212509728c8e199066b5e2

File: deploy/docker-compose.nonroot.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# !! If you use a .env file, use single quotes instead of escaping !!
+
+version: "3"
+services:
+  timetagger:
+    image: ghcr.io/almarklein/timetagger:v23.9.2-nonroot
+    ports:
+      - "80:80"
+    volumes:
+      - ./_timetagger:/opt/timetagger/_timetagger
```
[the remark this answers is not in the document]
Comment (kraftner) [reply deploy/docker-compose.nonroot.yml]: see https://github.com/almarklein/timetagger/pull/410/commits/4dca6c13c8e8e15de4c52fbd7c4c9d50363f4dff

**Comment by kraftner** [comment]
@danielhass I indeed left out the migration instructions and a more detailed explainer for now since I wanted to talk to you about it first.

I think we now have the bare minimum with an example and a one-liner explaining the difference. One thing that I think would maybe also be nice to tackle at the same time is #411 since that still makes the nonroot variant a bit clunky to use.

Next step could be to either merge this PR or extend it with the bigger documentation rework. What do you think @almarklein?

**Comment by almarklein** [comment]
Awesome! 

> Next step could be to either merge this PR or extend it with the bigger documentation rework. What do you think @almarklein?

I've merged this PR, since this is now in a good state. Thanks!

**Pull request merged**